### ISSUE TITLE
Add syntax customizable highlighting for generic placeholders

### ISSUE THREAD

**Issue by EmilySeville7cfg** [InBrowserApp/tldr.inbrowser.app#2]
Add syntax customizable highlighting for generic placeholders:

- strings:
  - default: highlight quotes and content inside them with orange color
  - settings: quotes and content color can be changed just together
- numbers
  - default: highlight integers and floats in cyan color
  - settings: integer and float colors can be changed just together
- ranges
  - default:
    - highlight borders (integer or float numbers) in cyan
    - highlight `-infinity` and `infinity` keywords in blue
    - highlight `..` in light cyan
  - settings: can be customized for 3 different aspects separately
- path placeholders
  - default: highlight paths in red, while highlighting their separators in dark red
  - settings: can be customized for 3 different aspects separately
- placeholders with ellipsis
  - defaults:
    - all defaults listed above applies for ellipsis placeholders
    - `...` is highlighted as gray
  - settings: only changing `...` color is supported, all other settings are being set for concrete placeholder types

**Comment by EmilySeville7cfg** [comment]
Yeah, our official Python client supports highlighting, but in a much simplified manner: just placeholder content is emphasized (without respect of the kind of it).

![image](https://user-images.githubusercontent.com/42812113/210572160-a4843801-263f-4600-9b68-116acab30b9f.png)

**Comment by rwv** [comment]
python color code:

https://github.com/tldr-pages/tldr-python-client/blob/c7d70091e958d1ccddd1e038578ad9ee897c5358/tldr.py#L344-L369

**Comment by EmilySeville7cfg** [comment]
I think that a proper parser can be written to process placeholders. We already have `tldr-lint`, check it out how pages are analyzed for smells, maybe you find smth useful. But for the beginning regex based highlighting can be used.

**Comment by rwv** [comment]
https://syntax-highlighting.tldr-inbrowser-app.pages.dev/

**Comment by EmilySeville7cfg** [comment]
I am glad to see syntax highlighting enabled. 😄
